### PR TITLE
Improve some checks for distributed Triangulations.

### PR DESCRIPTION
**Pull request by drwells**
Related to #12287 - we can improve these checks now that we have a common base class for distributed triangulations.

### CODE REVIEW

**Review by marcfehling — APPROVED** [review]

File: include/deal.II/numerics/error_estimator.templates.h [hunk 1]
```diff
@@ -1160,30 +1160,23 @@ KellyErrorEstimator<dim, spacedim>::estimate(
...
-       subdomain_id_);
-#else
-  const types::subdomain_id subdomain_id = subdomain_id_;
-#endif
+  types::subdomain_id subdomain_id = numbers::invalid_subdomain_id;
+  if (const auto *triangulation = dynamic_cast<
+        const parallel::DistributedTriangulationBase<dim, spacedim> *>(
+        &dof_handler.get_triangulation()))
+    {
+      Assert((subdomain_id == numbers::invalid_subdomain_id) ||
```
Comment (marcfehling) [inline include/deal.II/numerics/error_estimator.templates.h:1168]: Isn't this check redundant as you set the variable to this specific value above?

**Comment by marcfehling** [comment]
We can improve these checks even further once  #12287 is merged.

File: include/deal.II/numerics/error_estimator.templates.h (hunk 1 above)
Comment (drwells) [reply include/deal.II/numerics/error_estimator.templates.h:1168]: I checked the wrong thing - those should be for the input parameter `subdomain_id_` and not `subdomain_id`.

**Comment by drwells** [comment]
We can, but it is going to be tricky to do that in a backwards-compatible way since we still technically support parallelization with plain old `T` by giving each processor a distinct subdomain id.

**Review by marcfehling — APPROVED** [review]

**Pull request merged**